src/lib/search.js

```diff
@@ -12,6 +12,11 @@
 var isNumeric = require('fast-isnumeric');
 var loggers = require('./loggers');
 
+// don't trust floating point equality - fraction of bin size to call
+// "on the line" and ensure that they go the right way specified by
+// linelow
+var roundingError = 1e-9;
```

Comment:
I'm a little surprised that this hadn't come up before (other than the test [below](https://github.com/plotly/plotly.js/pull/2113/commits/6dc99047d171ac7acea56d5471cb0cbdff2ecdb2#diff-8c89958f1286cac36ab2325921671724R426) that was actually wrong before), and obviously it's a bit questionable exactly what small fraction to put here... but if we just use straight `<` vs `<=` etc it's pretty easy to trick `findBin` on the edges.